libs/wingsdk/src/target-sim/util.ts

```diff
@@ -50,13 +50,35 @@ export function bindSimulatorResource(
 export function makeSimulatorJsClient(filename: string, resource: Resource) {
   const type = basename(filename).split(".")[0];
   const env = makeEnvVarName(type, resource);
-  return `(function(env) {
-  let handle = process.env[env];
+
+  // return an object where calling any method will make a request to the simulator server
+  return `(function() {
+  const handle = process.env["${env}"];
   if (!handle) {
     throw new Error("Missing environment variable: " + env);
   }
-  return $simulator.findInstance(handle);
-})("${env}")`;
+  const simulatorUrl = process.env.WING_SIMULATOR_URL;
+  if (!simulatorUrl) {
+    throw new Error("Missing environment variable: WING_SIMULATOR_URL");
+  }
+
+  return new Proxy({}, {
+    get: function(target, method, receiver) {
+      return async function() {
+        const resp = await fetch(simulatorUrl + "/v1/call", {
+          method: "POST",
+          headers: { "Content-Type": "application/json" },
+          body: JSON.stringify({ handle, method, args: Array.from(arguments) }),
```

Comment:
We're subject to the JSON stringify limitations here. Some data types like BigInt, Date, etc won't work. I'd suggest using https://github.com/blitz-js/superjson instead.

Reply:
Sounds like a good fit, I'll try it out

Reply:
If the goal is to more closely resemble real cloud clients in the sim, should we not keep the same limitations that plain old JSON APIs force on us?

Reply:
> If the goal is to more closely resemble real cloud clients in the sim, should we not keep the same limitations that a plain old JSON APIs force on us?

I think we should talk about this, it's important. The initial solution to this problem was to change the way sim clients are implemented: they should be aware that they're being executed in a server and the communication protocol is HTTP.

The problem is that this is not the case currently. The quick solution was to create an HTTP server as a means of an invisible proxy for other processes to interact with the sim clients that currently exist...

Reply:
But this HTTP proxy comes with the problem that it can't serialize everything: Date, BigInt, Blob, and more importantly, Functions passed as callbacks, and other stuff that may be implemented in inflight methods.

Reply:
I believe the protocol used by our HTTP clients to send data to/from the process performing the simulation is just an implementation detail -- it's no different from how AWS SDK clients could be sending data to AWS through protobuf or via JSON strings.

Reply:
It's not an implementation detail, it's part of the contract of an API. If we plan to expose it at any capacity then we should probably just use standard json.

If we don't plan to expose it and always intend to control both sides, then this is fine and there's no need to read this comment any further.

> it's no different from how AWS SDK clients could be sending data to AWS through protobuf or via JSON strings.

I'm not sure I understand the comparison, the raw JSON is what I'm concerned about. Regardless of the existence of a client that can take a `Date()`, the format that the JSON uses should ideally use a standard representation of the data. I think this would prevent surprises from hitting the endpoint without our client and/or when implementing a client in a new language.

Reply:
@skyrpex Definitely -- I think eventually it would be ideal support other kinds of values as arguments / return types in inflight APIs (like callbacks), but that won't be possible with this current HTTP server approach. This limitation is OK for now -- we can evolve our protocol later to support this.

@MarkMcCulloh you're right, it depends on whether we're defining the request/responses of HTTP requests as part of our public contract. For now let's not make it part of our public API contract (don't advertise it in our docs) and tell anyone that asks that it may break at any point. After some time / once there's a need,  we can choose to make it a public contract.

Reply:
@skyrpex I did try adding superjson support in this PR -- unfortunately it's not working because of an issue in our simulator. https://github.com/winglang/wing/issues/4131

I'm not quite sure what the root cause is so I'll probably go back to the previous hack for now